test/test_run.py

```diff
@@ -45,27 +41,18 @@ def hdf_storage_factory():
     return SampleStorageHDF(file_path=os.path.join(work_dir, "mlmc_test.hdf5"))
 
 
-# Create sampling pools
-sampling_pool_single_process = OneProcessPool()
-sampling_pool_processes = ProcessPool(4)
-sampling_pool_thread = ThreadPool(4)
-sampling_pool_single_process_dir = OneProcessPool(work_dir=work_dir)
-sampling_pool_processes_dir = ProcessPool(4, work_dir=work_dir)
-sampling_pool_thread_dir = ThreadPool(4, work_dir=work_dir)
-
-
-@pytest.mark.parametrize("test_case", [(simulation, storage_memory, sampling_pool_single_process),
-                                       (simulation, storage_memory, sampling_pool_processes),
-                                       # (simulation, storage_memory, sampling_pool_thread),
-                                       (simulation, hdf_storage_factory(), sampling_pool_single_process),
-                                       (simulation, hdf_storage_factory(), sampling_pool_processes),
-                                       #(simulation, hdf_storage_factory(), sampling_pool_thread),
-                                       (simulation_workspace, storage_memory, sampling_pool_single_process_dir),
-                                       (simulation_workspace, storage_memory, sampling_pool_processes_dir),
-                                       #(simulation_workspace, storage_memory, sampling_pool_thread_dir),
-                                       (simulation_workspace, hdf_storage_factory(), sampling_pool_single_process_dir),
-                                       (simulation_workspace, hdf_storage_factory(), sampling_pool_processes_dir),
-                                       #(simulation_workspace, hdf_storage_factory(), sampling_pool_thread_dir)
+@pytest.mark.parametrize("test_case", [(SynthSimulationForTests(simulation_config), Memory(), OneProcessPool()),
```

Comment:
To preserve clarity one can decalre a factory functions:
```
def sampling_pool_single_process():
   return OneProcessPool()
```
... only meaningful for more parameters.